### PR TITLE
Remove RWLock from EntryNotifier because it causes perf degradation

### PR DESCRIPTION
**Pull request by vovkman**
#### Problem
When upgrading to 1.16 we observed that some of our geyser nodes could not keep up with the chain and had wild CPU fluctuations (running on 7443p). The first solution we found was to increase CPU (7543p), which sort of solved the issue. We profiled a local cluster to find any hotspots and found that the EntryNotifier was using significantly more CPU time than any other Geyser method. It seemed to be due to some thread contention from recv_timeout or the write lock is was obtaining. As a simple first step I removed the RWLock on EntryNotifier because it is unnecessary and re profiled. This removed the CPU overhead. I tested this change on MB with our geyser plugins and they are now stable on 7443p again.

Here is an example running yellowstone grpc plugin. This is from running a local cluster for about 10 minutes and shows he perf improvement. On MB the impact is much more severe, I didn't profile there though due to time/perf data size. Since a test of this change on MB fixed the issue I felt this was enough to illustrate the problem/solution.

**With RWLock - 0.234% CPU time** 
Screenshot 2023-10-19 at 4 34 04 PM

**Without RWLock - 0.147% CPU time**
Screenshot 2023-10-19 at 4 28 38 PM

#### Summary of Changes

Remove RWLock from EntryNotifier

### CODE REVIEW

File: ledger/src/entry_notifier_interface.rs [hunk 1]
```diff
@@ -8,4 +8,4 @@ pub trait EntryNotifier {
     fn notify_entry(&self, slot: Slot, index: usize, entry: &EntrySummary);
 }
 
-pub type EntryNotifierLock = Arc<RwLock<dyn EntryNotifier + Sync + Send>>;
```
Comment (lijunwangs) [inline ledger/src/entry_notifier_interface.rs:11]: Please remove the unused RwLock to make sure clippy is happy

Comment (vovkman) [reply ledger/src/entry_notifier_interface.rs:11]: done!

**Comment by vovkman** [comment]
@lijunwangs do these remaining buildkite steps need to pass? They seem to be pulling in the wrong dependencies and failing bc of that

**Comment by CriesofCarrots** [comment]
> @lijunwangs do these remaining buildkite steps need to pass? They seem to be pulling in the wrong dependencies and failing bc of that

That failure seems legitimate, and not related to dependencies:
```
Checking solana-geyser-plugin-manager v1.18.0 (/solana/geyser-plugin-manager)
error[E0277]: the trait bound `std::sync::RwLock<EntryNotifierImpl>: EntryNotifier` is not satisfied
   --> geyser-plugin-manager/src/geyser_plugin_service.rs:105:18
    |
105 |             Some(Arc::new(RwLock::new(entry_notifier)))
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `EntryNotifier` is not implemented for `std::sync::RwLock<EntryNotifierImpl>`
    = help: the trait `EntryNotifier` is implemented for `EntryNotifierImpl`
    = note: required for the cast from `Arc<std::sync::RwLock<EntryNotifierImpl>>` to `Arc<dyn EntryNotifier + std::marker::Send + Sync>`

For more information about this error, try `rustc --explain E0277`.
could not compile `solana-geyser-plugin-manager` (lib) due to previous error
```

**Comment by vovkman** [comment]
> > @lijunwangs do these remaining buildkite steps need to pass? They seem to be pulling in the wrong dependencies and failing bc of that
> 
> Those failures seem legitimate, and not related to dependencies:
> 
> ```
> Checking solana-geyser-plugin-manager v1.18.0 (/solana/geyser-plugin-manager)
> error[E0277]: the trait bound `std::sync::RwLock<EntryNotifierImpl>: EntryNotifier` is not satisfied
>    --> geyser-plugin-manager/src/geyser_plugin_service.rs:105:18
>     |
> 105 |             Some(Arc::new(RwLock::new(entry_notifier)))
>     |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `EntryNotifier` is not implemented for `std::sync::RwLock<EntryNotifierImpl>`
>     = help: the trait `EntryNotifier` is implemented for `EntryNotifierImpl`
>     = note: required for the cast from `Arc<std::sync::RwLock<EntryNotifierImpl>>` to `Arc<dyn EntryNotifier + std::marker::Send + Sync>`
> 
> For more information about this error, try `rustc --explain E0277`.
> could not compile `solana-geyser-plugin-manager` (lib) due to previous error
> ```

Oh rip I must have missed a push. Updated, all passing now

**Review by lijunwangs — APPROVED** [review]
Looks good

**Review by lijunwangs — APPROVED** [review]

**Pull request merged**